### PR TITLE
Pylab example moves 2

### PR DESCRIPTION
**Pull request by dstansby**
Some more pylab example moves. Trying to keep these in relatively small PRs for easy review.

### CODE REVIEW

File: examples/misc/table_demo.py [hunk 1]
```diff
@@ -9,11 +9,11 @@
 import matplotlib.pyplot as plt
 
 
-data = [[  66386,  174296,   75131,  577908,   32015],
```
Comment (tacaswell) [inline examples/misc/table_demo.py]: :bike: :derelict_house:  minor preference for leaving this formatting the way it is for human readability.

File: examples/pie_and_polar_charts/pie_demo2.py [hunk 2]
```diff
@@ -4,7 +4,8 @@
 =========
 
 Make a pie charts of varying size - see
-http://matplotlib.org/api/pyplot_api.html#matplotlib.pyplot.pie for the docstring.
+http://matplotlib.org/api/pyplot_api.html#matplotlib.pyplot.pie for the
```
Comment (tacaswell) [inline examples/pie_and_polar_charts/pie_demo2.py]: https

File: examples/scales/log_bar.py [hunk 3]
```diff
@@ -3,26 +3,24 @@
...
 =======
 
+Plotting a bar chart with a logarithmic y-axis.
 """
 import matplotlib.pyplot as plt
 import numpy as np
 
 data = ((3, 1000), (10, 3), (100, 30), (500, 800), (50, 1))
 
-plt.xlabel("FOO")
```
Comment (tacaswell) [inline examples/scales/log_bar.py]: Can we just change these to not 'FOO'?  It is good practice to always label your axes :smiling_imp:

**Comment by dstansby** [comment]
Comments fixed. Can always squash everything to a couple of commits before this is merged.

**Comment by tacaswell** [comment]
I am indifferent if this gets squashed or not.

File: examples/scales/log_demo.py [hunk 4]
```diff
@@ -13,19 +14,19 @@
...
 plt.subplot(221)
-plt.semilogy(t, np.exp(-t/5.0))
+plt.semilogy(t, np.exp(-t / 5.0))
 plt.title('semilogy')
 plt.grid(True)
 
 # log x axis
 plt.subplot(222)
-plt.semilogx(t, np.sin(2*np.pi*t))
+plt.semilogx(t, np.sin(2 * np.pi * t))
```
Comment (NelleV) [inline examples/scales/log_demo.py]: FYI, the old formatting was fine in terms of pep8ness.

**Comment by QuLogic** [comment]
Also needs a rebase, sorry.

**Comment by dstansby** [comment]
Should also be rebased, fingers crossed for the doc build!

**Pull request merged**